File: tests/typesafety/python/plumbing_by_hand.py
# expect: passes
# title: A loop accumulates the gift total by hand
from dataclasses import dataclass
from typing import NewType

Pence = NewType("Pence", int)


@dataclass(frozen=True)
class InvoiceLine:
    item: str
    pence: Pence
    gift: bool


def gift_total(lines: list[InvoiceLine]) -> Pence:
    total = 0
    for line in lines:
        if line.gift:
            total += line.pence
    return Pence(total)

File: tests/typesafety/typescript/plumbing_by_hand.ts
// expect: passes

type Order = { readonly item: string; readonly totalCents: number; readonly gift: boolean };

export function giftSpendCents(orders: Order[]): number {
  let sum = 0;
  for (const order of orders) {
    if (order.gift) {
      sum += order.totalCents;
    }
  }
  return sum;
}
